joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/CallTests.scala

```diff
@@ -18,13 +19,26 @@ class CallTests extends RubyCode2CpgFixture {
     puts.lineNumber shouldBe Some(2)
     puts.code shouldBe "puts 'hello'"
 
-    val List(rec: Identifier, hello: Literal) = puts.argument.l: @unchecked
-    rec.argumentIndex shouldBe 0
-    rec.name shouldBe "puts"
+    val List(selfReceiver: Identifier, hello: Literal) = puts.argument.l: @unchecked
+    selfReceiver.argumentIndex shouldBe 0
+    selfReceiver.name shouldBe RubyDefines.Self
+    selfReceiver.code shouldBe RubyDefines.Self
 
     hello.argumentIndex shouldBe 1
     hello.code shouldBe "'hello'"
+
+    val List(callBase: Call) = puts.receiver.l: @unchecked
+    callBase.argumentIndex shouldBe -1
+    callBase.methodFullName shouldBe Operators.fieldAccess
+    callBase.name shouldBe Operators.fieldAccess
+    callBase.code shouldBe "self.puts"
+
+    val List(baseSelf: Identifier, baseProperty: FieldIdentifier) = callBase.argument.l: @unchecked
+    baseSelf.argumentIndex shouldBe 1
+    baseSelf.name shouldBe RubyDefines.Self
     hello.lineNumber shouldBe Some(2)
```

Comment:
This check seems to be at an odd position now.

Reply:
This is modelled after what `self.foo('hello')` in PythonSrc would look like

Reply:
I meant that `hello.lineNumber ...` needs to go up where the rest of the `hello` related checks are.